reference.go

```diff
@@ -488,3 +488,43 @@ func ReferenceIsValidName(name string) bool {
 	}
 	return false
 }
+
+const (
+	// This should match GIT_REFNAME_MAX in src/refs.h
+	refnameMaxLength = 1024
+)
+
+type ReferenceFormat uint
+
+const (
+	ReferenceFormatNormal           ReferenceFormat = C.GIT_REFERENCE_FORMAT_NORMAL
+	ReferenceFormatAllowOnelevel    ReferenceFormat = C.GIT_REFERENCE_FORMAT_ALLOW_ONELEVEL
+	ReferenceFormatRefspecPattern   ReferenceFormat = C.GIT_REFERENCE_FORMAT_REFSPEC_PATTERN
+	ReferenceFormatRefspecShorthand ReferenceFormat = C.GIT_REFERENCE_FORMAT_REFSPEC_SHORTHAND
+)
+
+// ReferenceNormalizeName normalizes the reference name and checks validity.
+//
+// This will normalize the reference name by removing any leading slash '/'
+// characters and collapsing runs of adjacent slashes between name components
+// into a single slash.
+//
+// See git_reference_symbolic_create() for rules about valid names.
+func ReferenceNormalizeName(name string, flags ReferenceFormat) (string, error) {
+	cname := C.CString(name)
+	defer C.free(unsafe.Pointer(cname))
+
+	bufSize := C.size_t(refnameMaxLength)
```

Comment:
Since this is not gonna change, you could have it as a const.

```go
const (
	// This should match GIT_REFNAME_MAX in src/refs.h
	refnameMaxLength = 1024
	refnameBufSize = C.size_t(refnameMaxLength)
)
```

or just have it as:

```go
const (
	// This should match GIT_REFNAME_MAX in src/refs.h
	refnameBufSize = C.size_t(1024)
)
```